cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterSupplyUtils.java

```diff
@@ -158,4 +167,173 @@ public static WATER_USER_CONTRACT_TAB_T toWaterUserContractTs(WaterUserContract
+                    case BELOW:
+                        timeWindow.setLOCATION_REF(LocationUtil
+                                .getLocationRef(accounting.getPumpLocations().getPumpBelow()));
+                        break;
+                    default:
+                        LOGGER.log(Level.WARNING, "Invalid pump type");
+                        break;
+                }
+                timeWindow.setSTART_DATE(Timestamp.from(entry.getKey()));
+                timeWindow.setEND_DATE(Timestamp.from(entry.getKey()));
+                timeWindowList.add(timeWindow);
+            }
+        }
+        return new LOC_REF_TIME_WINDOW_TAB_T(timeWindowList);
+    }
+
+    public static List<WaterSupplyAccounting> toWaterSupplyAccountingList(Connection c, WAT_USR_CONTRACT_ACCT_TAB_T
+            watUsrContractAcctTabT) {
+
+        List<WaterSupplyAccounting> waterSupplyAccounting = new ArrayList<>();
+        Map<AccountingKey, WaterSupplyAccounting> cacheMap = new TreeMap<>();
+
+        for (WAT_USR_CONTRACT_ACCT_OBJ_T watUsrContractAcctObjT : watUsrContractAcctTabT) {
+            AccountingKey key = new AccountingKey.Builder()
+                .withContractName(watUsrContractAcctObjT.getWATER_USER_CONTRACT_REF().getCONTRACT_NAME())
+                .withWaterUser(new WaterUser.Builder()
+                    .withWaterRight(watUsrContractAcctObjT.getWATER_USER_CONTRACT_REF().getWATER_USER().getWATER_RIGHT())
+                    .withEntityName(watUsrContractAcctObjT.getWATER_USER_CONTRACT_REF().getWATER_USER().getENTITY_NAME())
+                    .withProjectId(CwmsId.buildCwmsId(watUsrContractAcctObjT
+                            .getWATER_USER_CONTRACT_REF().getWATER_USER().getPROJECT_LOCATION_REF().getOFFICE_ID(),
+                        watUsrContractAcctObjT.getWATER_USER_CONTRACT_REF().getWATER_USER().getPROJECT_LOCATION_REF().call_GET_LOCATION_ID()))
+                    .build())
+                .build();
+            if (cacheMap.containsKey(key)) {
+                WaterSupplyAccounting accounting = cacheMap.get(key);
+                addTransfer(watUsrContractAcctObjT, accounting);
+            } else {
+                cacheMap.put(key, createAccounting(c, watUsrContractAcctObjT));
+            }
+        }
+        for (Map.Entry<AccountingKey, WaterSupplyAccounting> entry : cacheMap.entrySet()) {
+            waterSupplyAccounting.add(entry.getValue());
+        }
+        return waterSupplyAccounting;
+    }
+
+    private static WaterSupplyAccounting createAccounting(Connection c, WAT_USR_CONTRACT_ACCT_OBJ_T acctObjT) {
+        WaterContractDao waterContractDao = new WaterContractDao(DSL.using(c));
+        WaterUserContract waterUserContract = waterContractDao.getWaterContract(
+            acctObjT.getWATER_USER_CONTRACT_REF().getCONTRACT_NAME(),
+            new CwmsId.Builder()
+                .withOfficeId(acctObjT.getWATER_USER_CONTRACT_REF().getWATER_USER().getPROJECT_LOCATION_REF().getOFFICE_ID())
```

Comment:
for readability, extract `acctObjT.getWATER_USER_CONTRACT_REF().getWATER_USER()` to a variable

Reply:
Extracted to variable